### ISSUE TITLE
Create a new documentation site (feedback wanted)

### ISSUE THREAD

**Issue by nknapp** [wycats/handlebars.js#1345]
I would like to create a new documentation site for the handlebarsjs.com and but I would like to hear some opinions. I think there are many things that can be made improved and there were several issues in the passed months due to users not finding the appropriate index.

What I would like to have is:

* A more detailed tutorial that leads the user from the basics to the advanced features (with a well visible trail to follow) 
* Some kind of navigation structure for users that seek help on a specific topic
* The ability to play with examples within the site
* (optional) A search function

Maybe we can use Handlebars itself to generate the site instead of the current engine. I don't know https://github.com/bootprint/bootprint can be used, but there are many static-site generators that support Handlebars.

I'd like to hear some feedback and I don't want to create such a site on my own. But if there are a couple of people willing to help, I would try to create a new repo to start with.

**Comment by jonschlinkert** [comment]
Hi @nknapp! I might have created an issue about this a while back, but I'd be happy to help. Would you be open for a pr to create a site similar to http://breakdance.io/? I can try to minimize indirection and abstractions to make it as easy as possible to maintain. I'm about to release new docs for [assemble](https://github.com/assemble/assemble), and since handlebars is a big part of our ecosystem this would align nicely and give me the motivation to finish that.

**Comment by nknapp** [comment]
Thanks for the offer. I have no access to the `handlebars-site`-repo yet, but I think I would start off in a new repository and on a different URL. I'm not sure where to start it, yet. Maybe I ask the people from https://github.com/handlebars-lang, if they are interested as well. Otherwise, I would create a new one.
I'll try to set up something tonight (MESZ), when I'm at home.

**Comment by jonschlinkert** [comment]
great, I'll start working on the bare bones structure, and I'll follow your lead regarding the details!

**Comment by nknapp** [comment]
@mmun Are you interested in a new documentation site for Handlebars, too? Personally, I think it would be good, if handlebars-related stuff (spec, docs etc) would be part of the same github-org. Would you be willing to give us the necessary access to the org?

@wycats Do you have opinion on creating a completely new `handlebarsjs.com` (it's your project after all)?

**Comment by mmun** [comment]
@nknapp The handlebars site definitely has some searchability issues that would be great to resolve.

Re: https://github.com/handlebars-lang. I made this org to house the work-in-progress spec, but it's a lot of work and it will be be some time before I can revisit it. Happy to use it to house other projects too!

**Comment by nknapp** [comment]
@mmun That's great. The easiest for me would be to become co-owner of the org, but it feels too much to ask for, given that you hardly know me. 

A team with a repo (`handlebars-site`?) would be enough to start with. Maybe just add @jonschlinkert and me (and you?) as team-members.

**Comment by mmun** [comment]
@nknapp Sure. I've made you a co-owner. I'm sure @wycats would appreciate if you wait for his sign-off before locking in any changes as well.

**Comment by nknapp** [comment]
Thanks @mmun. I'm not sure I understand what you mean by "locking in any changes". Do you mean "creating a new repository for documentation"? Or do you mean "creating a site that looks like an official Handlebars-site"?

Because I feel motivated and I would like to start right now. Of course, the new site will not instantly replace the old one. @wycats will still be able to veto and I would not just replace the old site without him agreeing.

Oh well, may be waiting a couple of days will not hurt...

**Comment by nknapp** [comment]
I just chatted with @wycats and he said (among other things):

> I would be happy to have a big rethink
> just want an end-to-end starting point that covers all the ground

I'll create a repo then.

**Comment by mmun** [comment]
@nknapp Sorry, I just meant that you should sync up with Yehuda before you go too far down one road. It seems like you have! Looking forward to your ideas.

**Comment by jonschlinkert** [comment]
quick update, I started working on converting the existing docs to markdown as a starting point. Just wanted to bring that up now in case anyone has objections. I should be able to push something up this weekend when I have some free time to work on it.

**Comment by nknapp** [comment]
I thought about markdown as well. I wouldn't have used the existing docs though.
Maybe just use the starting page, because that already fulfills @wycats expectations. 

I would like to see how the general file structure would look like. I would like to separate out the examples into their own files.



Am 18. Mai 2017 01:45:13 MESZ schrieb Jon Schlinkert <notifications@github.com>:
>quick update, I started working on converting the existing docs to
>markdown as a starting point. Just wanted to bring that up now in case
>anyone has objections. I should be able to push something up this
>weekend when I have some free time to work on it. 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/wycats/handlebars.js/issues/1345#issuecomment-302261446

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by nknapp** [comment]
Question: Does asssemble support async helpers in any way, or could it use [promised-handlebars](https://github.com/nknapp/promised-handlebars)?

**Comment by jonschlinkert** [comment]
> I wouldn't have used the existing docs though.

Really won't make a difference in terms of time. I just ran [breakdance](http://breakdance.io/) on the files. I wouldn't do that manually, I'm too lazy lol. 

> Does asssemble support async helpers in any way, or could it use promised-handlebars?

yes, it does, but it supports any engine, so we can use that if you want. let me get @doowb's opinion on this. we've had a few edge cases, maybe you were able to overcome those.

**Comment by doowb** [comment]
Hi @nknapp, it's nice to meet you. regarding the async helpers, would you want to move that discussion to a different venue?

**Comment by jonschlinkert** [comment]
forgot to mention, @doowb is co-maintainer of assemble and he also maintains the async-helpers lib

**Comment by nknapp** [comment]
@doowb where do you suggest to discuss this?

-

**Comment by nknapp** [comment]
@jonschlinkert I'll just wait for your push. Doesn't make sense to start working on if before.
We should discuss further steps in the [issue-tracker of the docs](https://github.com/handlebars-lang/docs/issues)-repo than

**Comment by doowb** [comment]
@nknapp we use slack and screenhero a lot. If any issues come from that, then we'll use GitHub issues. If you send me (or @jonschlinkert) an email (from my profile), we can get that setup.

**Comment by LarryAzevedo** [comment]
@nknapp, @jonschlinkert, @mmun, and @doowb, would you be interested in participating in a [card sorting](https://www.optimalworkshop.com/101/card-sorting) exercise?  It would help us discover how people understand and categorize handlebars content.

**Comment by nknapp** [comment]
I have created a new draft page at https://handlebars-lang.github.io/docs/, sources are on 
https://github.io/handlebars-lang/docs/

Anyone who wants to contribute content or styling is welcome to do so.

The new site is built with [vuepress](https://v1.vuepress.vuejs.org/), which is currently my favorite static site generator.

**Comment by LarryAzevedo** [comment]
@nknapp So I'm a little bit of a GitHub noob (I'm a UX Designer).  So how would I change something on https://handlebars-lang.github.io/docs/?  Do you have to give me permissions?

**Comment by nknapp** [comment]
I have added a CONTRIBUTING.md page that gives a little more detail about that. Usually, when you click the edit button on a repo you don't own, Github will  automatically create a fork and a pull request that I have to accept.

Or you click the "fork" button in the top right of the github-ui and it will create you own copy of the site repo and create a pull request after pushing changes to your fork.

I don't know about your general git and node skills. My UX skills are rather limited. It would certainly help if you open issues with change requests if you are not sure how to add them to the site.

**Comment by nknapp** [comment]
Let me know if this helps. I am interested in documenting the site itself so that everyone can contribute

**Comment by nknapp** [comment]
Oh. And I have never done a card sorting exercise. I'm not sure about my role in that. I rarely use the site myself. I still have limited resource and my primary goal for  now was to put d

**Comment by LarryAzevedo** [comment]
@nknapp Thanks for that info.  I will try it when I get a chance.

I looked into it more and card sorting would kinda be a pain to setup so everyone could do it.  Instead I made [some wireframes](http://larryazevedo.com/handlebars-wireframes.pdf) of what the site structure might look like.  Thoughts?

**Comment by LarryAzevedo** [comment]
@nknapp Any chance we can just do this with good old fashion HTML, CSS (maybe Sass), server-side code (like PHP) and call it a day?  I read [your reasons for using vuepress](https://github.com/handlebars-lang/docs#why-vuepress) but I think that's pigeonholing others because it forces people to know Vuepress and Stylus.  Which, a lot of people are not.  However, many do know HTML, CSS and Sass.

By the way, I'm not saying write all the CSS in one file with no variables, etc.  We can certainly leverage something like Sass to do some heavy lifting but give people technologies they're more likely to be familiar with.

Add yes, I'm willing to do the work on this.  I'm very comfortable writing semantic HTML, CSS that's responsive.  I'm just terrible when it comes to the developer setup part.

Thoughts?

**Comment by nknapp** [comment]
The site is hosted on github-pages (even the current handlebarsjs.com site). I'd like to keep it there, because its free and simple. But it means sticking to static sites.

For the most part, documenting stuff in vuepress comes down to just writing markdown, which is much easier to maintain than writing plain HTML.

Stylus can be omitted by writing a custom theme for vuepress. It's only the default theme that uses stylus. But changing to Sass probably means to create a completely new HTML from scratch.

**tldr;**

I'd like to stick with vuepress, but I am not opposed to creating a new theme with Sass. Still, I think the most important thing for now is getting the site filled with some content.

**Comment by LarryAzevedo** [comment]
@nknapp Ok we can stick with Vuepress and I agree, the most important thing is the content.  So I read your instructions but I don't see an _Edit_ button.  I just see this:

![Screen Shot 2019-04-29 at 10 42 57 AM](https://user-images.githubusercontent.com/2222751/56904162-bdcec380-6a6b-11e9-8b25-7dd1792cd817.png)

My guess was to click on _New pull request_ but I'm not sure.  I've never done this before but once I'm setup, I can really start to make progress.  If it's better for you to get on a Skype with me, that works (larry.azevedo).

**Comment by wycats** [comment]
@nknapp I'm 👍 on vuepress
